### PR TITLE
add metric for InMemoryDelayedDeliveryTracker's memory usage 

### PR DESCRIPTION
**Pull request by tjiuming**
### Motivation

Provide namespace level and topic level metric for `InMemoryDelayedDeliveryTracker's` memory usage.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [x] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by tjiuming** [comment]
Tests to be completed.

**Comment by tjiuming** [comment]
@codelipenghui  PTAL

File: pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/TopicStatsImpl.java [hunk 1]
```diff
@@ -131,6 +131,9 @@ public class TopicStatsImpl implements TopicStats {
     /** The serialized size of non-contiguous deleted messages ranges. */
     public int nonContiguousDeletedMessagesRangesSerializedSize;
 
+    /** The size of InMemoryDelayedDeliveryTracer memory usage. */
+    public int delayedTrackerMemoryUsage;
```
Comment (codelipenghui) [inline pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/TopicStatsImpl.java:135]:
```suggestion
    public int delayedMessageIndexSizeInBytes;
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java [hunk 2]
```diff
@@ -308,4 +307,70 @@ public void testAutoCreatePartitionedTopicThatNameIncludePartition() throws Exce
...
+            producer.newMessage().value(UUID.randomUUID().toString()).deliverAfter(30, TimeUnit.SECONDS).send();
+        }
+        producer.flush();
+
+        //wait for Dispatcher build delayed message index.
+        for (int a = 0; a < 100; a++) {
+            Thread.sleep(1);
+        }
+
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
```
[the remark this answers is not in the document]
Comment (tjiuming) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java:355]: no need to close it, it's memory managed by heap. also, ByteArrayOutputStream#close is noop.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java [hunk 3]
```diff
@@ -345,6 +346,9 @@ private static void printNamespaceStats(SimpleTextOutputStream stream, String cl
 
         metric(stream, cluster, namespace, "pulsar_subscription_delayed", stats.msgDelayed);
 
+        metric(stream, cluster, namespace, "pulsar_delayed_delivery_tracker_memory_usage",
```
[the remark this answers is not in the document]
Comment (tjiuming) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java:349]: it makes sense

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java [hunk 4]
```diff
@@ -308,4 +307,70 @@ public void testAutoCreatePartitionedTopicThatNameIncludePartition() throws Exce
...
+    public void testNamespaceLevelDelayedDeliveryTrackerMemoryUsageMetric() throws Exception {
+        String topic = "persistent://prop/autoNs/test_delayed_message_metric" + UUID.randomUUID();
+        testDelayedDeliveryTrackerMemoryUsageMetric(topic, false);
+    }
+
+    @Test
+    public void testTopicLevelDelayedDeliveryTrackerMemoryUsageMetric() throws Exception {
+        String topic = "persistent://prop/autoNs/test_delayed_message_metric" + UUID.randomUUID();
+        testDelayedDeliveryTrackerMemoryUsageMetric(topic, true);
+    }
```
[the remark this answers is not in the document]
Comment (tjiuming) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java:322]: could you please describe it more clearly?

[the remark this answers is not in the document]
Comment (codelipenghui) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/persistent/PersistentTopicTest.java:287]: I mean we don't need two methods, instead we can just create a data provider

e.g. https://github.com/apache/pulsar/blob/21c3a0b5c93fd30fd458fbab818d059fef4fd8ba/pulsar-io/elastic-search/src/test/java/org/apache/pulsar/io/elasticsearch/IndexNameFormatterTest.java#L45

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/AggregatedNamespaceStats.java [hunk 5]
```diff
@@ -59,6 +59,7 @@ public class AggregatedNamespaceStats {
     long compactionCompactedEntriesCount;
     long compactionCompactedEntriesSize;
     StatsBuckets compactionLatencyBuckets = new StatsBuckets(CompactionRecord.WRITE_LATENCY_BUCKETS_USEC);
+    public int delayedTrackerMemoryUsage;
```
Comment (Shoothzj) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/AggregatedNamespaceStats.java:62]:
```suggestion
    int delayedTrackerMemoryUsage;
```
I think we don't need `public`?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TopicStats.java [hunk 6]
```diff
@@ -67,6 +67,7 @@ class TopicStats {
     long compactionCompactedEntriesCount;
     long compactionCompactedEntriesSize;
     StatsBuckets compactionLatencyBuckets = new StatsBuckets(CompactionRecord.WRITE_LATENCY_BUCKETS_USEC);
+    public int delayedTrackerMemoryUsage;
```
Comment (Shoothzj) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TopicStats.java:70]:
```suggestion
    int delayedTrackerMemoryUsage;
```
do we really need public?

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@tjiuming Please check @Shoothzj 's comments

**Review by mattisonchao — APPROVED** [review]

**Comment by tjiuming** [comment]
@codelipenghui All CI checks passed, PTAL

**Pull request merged**